bin/migrate-oats-data/batch_applications.py

```diff
@@ -0,0 +1,143 @@
+from db import inject_conn_pool
+def application_etl_temp_table():
+    """
+    function creates a temporary table to be used by .sql files called later in script
+    """
+    return f"""
+        DROP TABLE IF EXISTS application_etl;
+
+        CREATE TEMPORARY TABLE application_etl (
+            id SERIAL PRIMARY KEY,
+            application_id int,
+            card_uuid UUID NOT NULL DEFAULT gen_random_uuid(),
+            duplicated bool DEFAULT false
+        )
+    """
+
+def application_etl_insert():
+    """
+    function inserts data into prevoisly created table and decipers duplication of uuids
+    application_id is copied from oats.oats_alr_applications.alr_application_id
+    """
+    return f"""
+        INSERT INTO
+            application_etl (application_id, duplicated)
+        SELECT
+            DISTINCT oa.alr_application_id AS application_id,
+            CASE
+                WHEN a.uuid IS NOT NULL THEN TRUE
+                ELSE false
+            END AS duplicated
+        FROM
+            oats.oats_alr_applications AS oa
+            LEFT JOIN alcs.application AS a ON oa.alr_application_id :: text = a.file_number
+    """
+# def associated_card():
+#     return f"""
+#         INSERT INTO alcs.card (uuid, audit_created_by)
+
+#         SELECT
+#             ae.card_uuid,
+#             'oats_etl'
+#         FROM
+#             application_etl ae
+#         WHERE
+#             ae.duplicated IS false;
+
+#     """
+
+def compile_application_insert_query(number_of_rows_to_insert):
+    """
+    function takes the number of rows to insert and generates an SQL insert statement with upserts using the ON CONFLICT clause
+    """
+
+    applications_to_insert = ",".join(["%s"] * number_of_rows_to_insert)
+    return f"""
+        INSERT INTO alcs.application (file_number, type_code, 
+                                      applicant, region_code, local_government_uuid, audit_created_by)
+
+        VALUES{applications_to_insert}
+        ON CONFLICT (file_number) DO UPDATE SET
```

Comment:
If you have not confirmed the "on conflict" condition with Ansley, let's do that. This condition needs to be approved by the business. Otherwise, we may be overriding the application with outdated info